File: CB.EN.U4CYS22030/FLOWGORITHM TO PYTHON/primechecker.py
# PRIME NUMBER CHECK
print("Enter the number to be checked : ")
n = int(input())
x = [0] * (n)

if n < 1:
    print("It's not possible for negative numbers")
else:
    if n == 0:
        print("Not a prime number")
    else:
        if n == 1:
            print("Not a prime number")
        else:
            if n == 2:
                print("It is a prime number")
            else:
                a = 2
                a1 = 0
                for i in range(n - 1, 2 - 1, -1):
                    if n % i == 0:
                        a1 = a1 + 1
                if a1 > 0:
                    print(str(n) + " is not a prime number ")
                    print("Number of divisors for " + str(n) + " is " + str(a + a1))
                else:
                    print(str(n) + " is a prime number ")
                    print("Number of divisors for " + str(n) + " is " + str(a + a1))

File: CB.EN.U4CYS22030/UBUNTU/prime_checker.c
#include<stdio.h>
int main(){
    int n;
    int a=0;
    printf("Enter the number to be checked : ");
    scanf("%d",&n);
    if(n<0){
	    printf("Negative numbers are not prime numbers\n");
    }
    else if(n==0){
	    printf("0 is not a prime number\n");
    }
    else if(n==1){
	    printf("Neither prime nor composite\n");
    }
    else if(n==2){
	    printf("2 is the only prime number\n");
    }
    else{
    for(int i=2;i<(n) && i>1;i++){
        if(n%i==0){
            a=a+1;
        }
    }
    }
    
    if(a>0){
        printf("%d is not a prime number.\n",n);
        printf("Number of factors for %d = %d\n",n,a+2);
    }
    else
    printf("%d is a prime number.\n",n);
}
